libutils/string_lib.c

```diff
@@ -231,7 +231,14 @@ int StringSafeCompare(const char *const a, const char *const b)
     }
     if (a != NULL && b != NULL)
     {
-        return strcmp(a, b);
+        // Adding this as strcmp gives difference of the buffer values at aarch64 whereas it gives 1 or -1 in other platforms, in case compared char buffer were not same.
+        int compare_result = strcmp(a, b);
+        if (compare_result == 0)
```

Comment:
Keep the same changes as you are doing in the file: libutils/buffer.c

Specially regarding this "if condition"